nixos/modules/services/computing/boinc/client.nix

```diff
@@ -0,0 +1,84 @@
+{config, lib, pkgs, ...}:
+
+with lib;
+
+let
+  cfg = config.services.boinc;
+  allowRemoteGuiRpcFlag = if cfg.allowRemoteGuiRpc then "--allow_remote_gui_rpc" else "";
+
+in
+  {
+    options.services.boinc = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        example = true;
+        description = ''
+          Whether to enable the BOINC distributed computing client. If this
+          option is set to true, the boinc_client daemon will be run as a
+          background service. The boinccmd command can be used to control the
+          daemon.
+        '';
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.boinc;
+        defaultText = "pkgs.boinc";
+        description = ''
+          Which BOINC package to use.
+        '';
+      };
+
+      dataDir = mkOption {
+        type = types.str;
+        default = "/var/lib/boinc";
+        description = ''
+          The directory in which to store BOINC's configuration and data files.
+        '';
+      };
+
+      allowRemoteGuiRpc = mkOption {
+        type = types.bool;
+        default = false;
+        example = true;
+        description = ''
+          If set to true, any remote host can connect to and control this BOINC
+          client (subject to password authentication). If instead set to false,
+          only the hosts listed in $dataDir/remote_hosts.cfg will be allowed to
+          connect.
+
+          See also: http://boinc.berkeley.edu/wiki/Controlling_BOINC_remotely#Remote_access
+        '';
+      };
+    };
+
+    config = mkIf cfg.enable {
+      environment.systemPackages = [cfg.package];
+
+      users.users.boinc = {
+        createHome = false;
+        description = "BOINC Client";
+        home = cfg.dataDir;
+        isSystemUser = true;
+      };
+
+      systemd.services.boinc = {
+        description = "BOINC Client";
+        requires = ["network.target"];
+        wantedBy = ["multi-user.target"];
+        script = ''
+          test -d ${cfg.dataDir} || mkdir -p ${cfg.dataDir}
```

Comment:
The 'test' is indeed redundant; accidentally left it in after debugging why the service was complaining about a non-existent data directory during startup (turns out ZFS filesystems aren't included in `local-fs.target`, and so aren't mounted by the time this service starts, but that's another issue).

The user has `createHome = false`, so it wouldn't be automatically trying to create the home directory. There's no reason why the home directory needs to be set to the data directory, it just seemed like a sensible default.

Either way, I've moved the directory creation into a `preStart` hook.